### PR TITLE
Addition of important terms to course schedule.

### PR DESCRIPTION
**Pull request by zaethovenn**
What I did was add definitions to chapters 2-16 to bring all of their definition counts to 5 per section. There are 15 following chapters that will need updating to kapfhammers 5 count per section request.

### CODE REVIEW

File: docs/operating-systems/schedule.md [hunk 1]
```diff
@@ -351,9 +351,17 @@ hide:
     - **Concurrency** - managed by the operating system and/or the programming
       langauge, the situation in which multiple computations happen at the same
       time
-    - **Persistence** - the process of storing data in a computer system, as a
+    - **Persistence** - the process of storing data in a computer system, as a 
       way of ensuring that data survives after the ending of the process that
       created it
+    - **journaling** - also known as copy-on-write, is a special ordering protocol
```
Comment (burgess01) [inline docs/operating-systems/schedule.md:357]: capitalize term to match the rest of the existing entries

File: docs/operating-systems/schedule.md [hunk 2]
```diff
@@ -401,6 +415,8 @@ hide:
     - **Stack Memory** - allocations and deallocations of it are managed implicitly by the compiler for you
     - **Buffer Overflow** - a related error is not allocating enough memory
     - **Purify and Valgrind** - both are excellent at helping you locate the source of your memory-related problems
+    - **Heap Memory** - all allocation and deallocation of heap memory is controlled soley by the programmer
+    - **malloc()** - you pass malloc a size you request for room on the heap that either passes you a pointer to your newly allocated space or fails and returns NULL
```
Comment (burgess01) [inline docs/operating-systems/schedule.md:419]: capitalize term to match the rest of the existing entries, also remove parenthesis as you don't include them for other function terms like fork

**Review by burgess01 — CHANGES_REQUESTED** [review]
There are some small formatting issues that I would like you to fix before this is merged. Also, for some reason, this PR does not have a preview build. If you would be able to get one so we can see how this will look when it is merged into main I would really appreciate that, thank you!

**Review by burgess01 — APPROVED** [review]
This looks great!

**Comment by gkapfham** [comment]
Note: This PR had the correct links for the semaphores content.

**Comment by gkapfham** [comment]
Note: this PR had the correct link for the semaphores content.

**Pull request merged**